scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/services/GraphQLMerlinService.java

```diff
@@ -803,7 +835,22 @@ private Map<SimulatedActivityId, SimulatedActivity> getSimulatedActivities(Simul
     return parseSimulatedActivities(data, startSimulation);
   }
 
-private Profiles getProfiles(DatasetId datasetId) throws PlanServiceException, IOException {
+  private ProfileSet getProfileTypes(DatasetId datasetId) throws PlanServiceException, IOException {
+    final var request = """
+        query{
+          profile(where: {dataset_id: {_eq: %d}}){
+            type
+            name
+          }
+        }
+        """.formatted(datasetId.id());
+    final JsonObject response;
+    response = postRequest(request).get();
+    final var data = response.getJsonObject("data").getJsonArray("profile");
+    return parseProfiles(data);
+  }
+
+private ProfileSet getProfilesWithSegments(DatasetId datasetId) throws PlanServiceException, IOException {
```

Comment:
Minor, can ignore: this line should be indented
```suggestion
  private ProfileSet getProfilesWithSegments(DatasetId datasetId) throws PlanServiceException, IOException {
```